tests/parser/run-parser-tests.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+// Throw exception if anything fails.
+set_error_handler(
+	function ( $severity, $message, $file, $line ) {
+		throw new ErrorException( $message, 0, $severity, $file, $line );
+	}
+);
+
+require_once __DIR__ . '/../../wp-includes/mysql/class-mysql-lexer.php';
+require_once __DIR__ . '/../../wp-includes/parser/class-parser.php';
+
+function getStats( $total, $failures, $exceptions ) {
+	return sprintf(
+		'Total: %5d  |  Failures: %4d / %2d%%  |  Exceptions: %4d / %2d%%',
+		$total,
+		$failures,
+		$failures / $total * 100,
+		$exceptions,
+		$exceptions / $total * 100
+	);
+}
+
+$grammar_data = include __DIR__ . '/../../wp-includes/mysql/mysql-grammar.php';
+$grammar      = new Grammar( $grammar_data );
+
+$handle     = fopen( __DIR__ . '/data/queries.csv', 'r' );
```

Comment:
Could we use PHPUnit here and plug this into the CI pipeline? It could still read the queries off the `.csv` file. If it used a "data provider", PHPUnit would treat each query as a separate test even though there would only be a single test handler.